smithy-openapi/src/main/java/software/amazon/smithy/openapi/fromsmithy/OpenApiJsonSchemaMapper.java

```diff
@@ -100,12 +94,37 @@ public Schema.Builder updateSchema(Shape shape, Schema.Builder builder, JsonSche
             }
         }
 
+        if (shape.hasTrait(BoxTrait.class)) {
+            handleNullability(builder, config);
+        }
+
         // Remove unsupported JSON Schema keywords.
         UNSUPPORTED_KEYWORD_DIRECTIVES.forEach(builder::disableProperty);
 
         return builder;
     }
 
+    private void handleFormatKeyword(Schema.Builder builder, OpenApiConfig config) {
+        String blobFormat = config.getDefaultBlobFormat();
+        if (config.getVersion().usesFormatKeywordForFilePayload()) {
+            builder.format(blobFormat);
+        } else {
+            builder.contentEncoding(blobFormat);
+        }
+    }
+
+    private void handleNullability(Schema.Builder builder, JsonSchemaConfig config) {
+        if (config instanceof OpenApiConfig
```

Comment:
Ah yeah, the name `supportsNullability` is too ambiguous. It could easily mean either branch here. Let's use a name like `supportsNullable` which is supported in 3.0.2 but not 3.1.0 (I think).

Reply:
Name changed to `supportsNullableKeyword` for removing ambiguity in name.